### PR TITLE
CI change for mergifyio `strict` mode deprecation

### PR DESCRIPTION
**Pull request by humblec**
ci: move from merge action to queue action
    
    as mentioned in the below blog the support for strict mode
    and merge action will be done soon in mergify. This brings
    the change requested for the same.
    
    Ref# https://blog.mergify.com/strict-mode-deprecation/

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Review by nixpanic — CHANGES_REQUESTED** [review]
Have you tested this in an other repository already? The multi-queue configuration requires to have the premium feature plan... We need to setup a single queue instead.

https://docs.mergify.com/actions/queue/#examples

**Comment by humblec** [comment]
> Have you tested this in an other repository already? The multi-queue configuration requires to have the premium feature plan... We need to setup a single queue instead.
> 
> https://docs.mergify.com/actions/queue/#examples

@nixpanic thanks.. missed this `plan` thing.. will revisit this PR for the single queue option then. looks difficult as the conditions are different for each branch. but will give a try

**Comment by Madhu-1** [comment]
>The configuration uses the deprecated strict mode of the merge action.
A brownout is planned for the whole December 6th, 2021 day.
This option will be removed on January 10th, 2022.
For more information: https://blog.mergify.com/strict-mode-deprecation/

@humblec this is needed as mergify configuration is not considered from today, can you please check on this one?

**Comment by humblec** [comment]
Yeah, today is the deprecation. looks like we are in that. eventhough its still 10th :)

**Comment by humblec** [comment]
hmm.. the error is confusing in `summary` check:

It is giving:
```


    The configuration uses the deprecated strict mode of the merge action.
    A brownout is planned for the whole December 6th, 2021 day.
    This option will be removed on January 10th, 2022.
    For more information: https://blog.mergify.com/strict-mode-deprecation/

rebase is invalid for dictionary value @ defaults → actions → merge → strict_method

    The configuration uses the deprecated strict mode of the merge action.
    A brownout is planned for the whole December 6th, 2021 day.
    This option will be removed on January 10th, 2022.
    For more information: https://blog.mergify.com/strict-mode-deprecation/

smart is invalid for dictionary value @ defaults → actions → merge → strict

```

The `strict` or `strict_method` is not part of the rules any more.

Can I have second pair of eyes on this @ceph/ceph-csi-maintainers

**Comment by Madhu-1** [comment]
https://dashboard.mergify.com/github/ceph/repo/ceph-csi/config-editor configurations look valid to me.

File: .mergify.yml
Comment (Rakshith-R) [inline .mergify.yml]:
```suggestion
      - or:
        - and:
      - "status-success=codespell"
      - "status-success=multi-arch-build"
      - "status-success=go-test"
      - "status-success=golangci-lint"
      - "status-success=commitlint"
      - "status-success=mod-check"
      - "status-success=lint-extras"
      - "status-success=ci/centos/k8s-e2e-external-storage/1.21"
      - "status-success=ci/centos/k8s-e2e-external-storage/1.22"
      - "status-success=ci/centos/mini-e2e-helm/k8s-1.20"
      - "status-success=ci/centos/mini-e2e-helm/k8s-1.21"
      - "status-success=ci/centos/mini-e2e-helm/k8s-1.22"
      - "status-success=ci/centos/mini-e2e-helm/k8s-1.23"
      - "status-success=ci/centos/mini-e2e/k8s-1.20"
      - "status-success=ci/centos/mini-e2e/k8s-1.21"
      - "status-success=ci/centos/mini-e2e/k8s-1.22"
      - "status-success=ci/centos/mini-e2e/k8s-1.23"
      - "status-success=ci/centos/upgrade-tests-cephfs"
      - "status-success=ci/centos/upgrade-tests-rbd"
      - "status-success=DCO"
```
Will not work for ci/centos branch, 
I'll push change into this pr in a min

File: .mergify.yml
Comment (Madhu-1) [inline .mergify.yml:16]: this might not work for the release-3.4 branch as we don't run k8s-e2e-external-storage tests on it and this won't work for the centos/ci branch as we don't run any of these tests on it.

**Comment by humblec** [comment]
> https://dashboard.mergify.com/github/ceph/repo/ceph-csi/config-editor configurations look valid to me.

yeah, but the summary test gives `invalid` . was confused.

File: .mergify.yml
Comment (humblec) [reply .mergify.yml]: ah.. yes, ci branch need changes. @Rakshith-R please feel free to push the changes :+1:

File: .mergify.yml
Comment (humblec) [reply .mergify.yml:16]: as @Rakshith-R  mentioned, centos ci branch need changes. will wait for the PR refresh, it should be covered in it.

**Comment by humblec** [comment]
By the proposed changes of conditions, this looks good now.. ptal thanks..

**Comment by humblec** [comment]
@Madhu-1 ptal ..

File: .mergify.yml
Comment (Madhu-1) [inline .mergify.yml]: duplicate check

File: .mergify.yml
Comment (Madhu-1) [inline .mergify.yml]: this is not required

File: .mergify.yml
Comment (Madhu-1) [inline .mergify.yml]: this check is not required.

**Review by Madhu-1 — CHANGES_REQUESTED** [review]
@humblec are these mergify changes validated with some test repository?

File: .mergify.yml
Comment (humblec) [reply .mergify.yml]: removed

File: .mergify.yml
Comment (humblec) [reply .mergify.yml]: Done

**Comment by humblec** [comment]
@Madhu-1 ptal.. thanks.

**Comment by humblec** [comment]
> @humblec are these mergify changes validated with some test repository?

no @Madhu-1

File: .mergify.yml
Comment (Madhu-1) [inline .mergify.yml]: looks like this is not removed yet.

Comment (humblec) [reply .mergify.yml]: should be good now @Madhu-1

**Review by Madhu-1 — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

**Review by nixpanic — APPROVED** [review]

**Pull request merged**
